libs/components/popovers/testing/src/dropdown/harness/dropdown-menu-harness.ts

```
@@ -0,0 +1,116 @@
+import { SkyDropdownMenuHarnessFilters } from './dropdown-menu-harness.filters';
+
+/**
+ * Harness for interacting with dropdown menu in tests
+ */
+export class SkyDropdownMenuHarness extends SkyComponentHarness {
+  /**
+   * @internal
+   */
+  public static hostSelector = '.sky-dropdown-menu';
+
+  #getDropdownItems = this.locatorForAll(SkyDropdownItemHarness);
+
+  /**
+   * Gets a `HarnessPredicate` that can be used to search for a
+   * `SkyDropdownMenuHarness` that meets certain criteria
+   */
+  public static with(
+    filters: SkyDropdownMenuHarnessFilters
+  ): HarnessPredicate<SkyDropdownMenuHarness> {
+    return SkyDropdownMenuHarness.getDataSkyIdPredicate(filters);
+  }
+
+  /**
+   * Gets the aria-labelledby value
+   */
+  public async getAriaLabelledBy(): Promise<string | null> {
+    return (await this.host()).getAttribute('aria-labelledby');
+  }
+
+  /**
+   * Clicks the dropdown menu item at index
+   * @param index Index of dropdown menu item
+   */
+  public async clickItemAtIndex(index: number): Promise<void> {
+    await (await this.getItemAtIndex(index))?.click();
+  }
+
+  /**
+   * Clicks the dropdown menu item with a specific role
+   * @param role Role of dropdown menu item to be clicked
+   */
+  public async clickItemWithRole(role: string): Promise<void> {
+    const menuItem = await this.getItemWithRole(role);
+    if (!menuItem) {
+      throw new Error(
+        `Unable to click item. Item with role ${role} does not exist in this dropdown menu`
+      );
+    }
+    await (await this.getItemWithRole(role))?.click();
+  }
+
+  /**
+   * Gets the dropdown menu item at a specific index
+   * @param index Index of dropdown menu item to be clicked
+   */
+  public async getItemAtIndex(
+    index: number
+  ): Promise<SkyDropdownItemHarness | undefined> {
+    const itemsList = await this.getItems();
+
+    if (itemsList?.length === 0) {
+      throw new Error('Unable to retrieve item because dropdown menu is empty');
```

Should we throw a different error if the menu is empty, versus not open?
`'Unable to retrieve an item because the dropdown menu is empty.'`
`'Unable to retrieve an item because the dropdown menu is not open.'`

---

I have that check in the `dropdown-harness.ts` and the wording for that error is `Unable to retrieve dropdown menu harness because dropdown is closed`